server/tests/unit/services/RunService.test.js

```diff
@@ -236,6 +237,85 @@ describe('RunService', () => {
+                        git_hash: process.env.IMPORT_ARIA_AT_TESTS_COMMIT_1
+                    }
+                });
+                const apgExamples = await db.ApgExample.findAll({
+                    where: { test_version_id: testVersion.id }
+                });
+
+                const apgExampleIds = apgExamples.map(example => example.dataValues.id)
+
+                const at = await db.At.findOne({
+                    where: { test_version_id: testVersion.id },
+                    include: [db.AtName]
+                });
+
+                const browser = await db.Browser.findOne();
+
+                const runStatus = await db.RunStatus.findOne({
+                    where: { name: db.RunStatus.RAW }
+                });
+
+                const activeRuns = await RunService.configureRuns({
+                    test_version_id: testVersion.id,
+                    apg_example_ids: apgExampleIds,
+                    at_browser_pairs: [
+                        {
+                            at_name_id: at.at_name_id,
+                            at_version: atVersionNumber,
+                            browser_id: browser.id,
+                            browser_version: browserVersionNumber
+                        }
+                    ]
+                });
+
+                // There are three apg examples
+                const keys = Object.keys(activeRuns);
+                expect(keys.length).toEqual(3);
+
+                // TODO: is there a better way to assert this?
+                Object.values(activeRuns).forEach((run, i) => {
+                    delete run.id
+                    let apgExample = apgExamples[i]
+                    let result = {
+                        browser_id: browser.id,
+                        browser_version: browserVersionNumber,
+                        browser_name: browser.name,
+                        at_id: at.id,
+                        at_key: at.key,
+                        at_name: at.AtName.name,
+                        at_version: atVersionNumber,
+                        apg_example_directory: apgExample.directory,
+                        apg_example_name: apgExample.name,
+                        apg_example_id: apgExample.id,
+                        run_status_id: runStatus.id,
+                        run_status: db.RunStatus.RAW,
+                        test_version_id: testVersion.id,
+                        testers: []
+                    }
+
+                    expect(run).toEqual(result);
+                });
+            });
+        });
+        it('creates several runs from a list of tech pairs', async () => {
+            //TODO
+        });
+        it('creates several runs from a list of apg examples and a list of tech pairs', async () => {
+            //TODO
+        });
+        it('creates several runs from a list of apg examples when there are existing apg examples in the database', async () => {
+            //TODO
+        });
```

Comment:
I don't think this is the correct list of tests that need to be written. (Not sure if is meant to be the full list or not). 

It doesn't add much to do a list of several things verses one of the things. In most cases that isn't a huge problem. I would probably only add tests like that if there was a specific bug. Or maybe one that doesn't cover all cases and just make sure that I have one test that covered all cahses. 

I think it would take me a while to sit down and think of all the cases that need to be covered. But the thing that doens't seem to be covered by these test case descriptions is the change of state... I don't think the creation of new state is probably that hard or a thing I'm really worried about catching bugs in but the thing I'm really nervous about and want throughly tested is when when some things already exist in the DB from the previously active run or the previously previouslu active run and we need to copy over some of thata state.